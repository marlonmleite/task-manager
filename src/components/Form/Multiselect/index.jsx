import PropTypes from 'prop-types'
import withFormGroup from '../withFormGroup'
import MultiselectComponent from './styled'

MultiselectComponent.defaultProps = {
  isMulti: true,
  classNamePrefix: 'multiselect',
}

MultiselectComponent.propTypes = {
  isMulti: PropTypes.bool,
}

const Multiselect = withFormGroup(MultiselectComponent)

export {
  MultiselectComponent,
  Multiselect,
}
