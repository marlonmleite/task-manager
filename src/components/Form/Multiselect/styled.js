import Select from 'react-select'
import styled from 'styled-components'

const SelectStyled = styled(Select)`
  .multiselect {
    &__control {
      border-color: #e5e5e5;
      border-style: solid;
      border-width: 0 0 1px 0;
      border-radius: 0;
      background-color: transparent;
      min-height: 38px;

      &--is-focused {
        box-shadow: none;
        border-color: ${({ theme }) => theme.focus};
      }
    }

    &__value-container {
      padding: 5px 5px 5px 0;
    }

    &__menu {
      border-radius: 2px;
    }
  }
`

export default SelectStyled
