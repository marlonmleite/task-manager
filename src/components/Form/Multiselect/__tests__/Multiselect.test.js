import React from 'react'
import { shallow } from 'enzyme'
import Multiselect from '../styled'

describe('Multiselect', () => {
  it('should render', () => {
    const select = shallow(<Multiselect />)

    expect(select).toMatchSnapshot()
  })
})
