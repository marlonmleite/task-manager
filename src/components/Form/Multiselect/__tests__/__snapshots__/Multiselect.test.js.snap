// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiselect should render 1`] = `
.c0 .multiselect__control {
  border-color: #e5e5e5;
  border-style: solid;
  border-width: 0 0 1px 0;
  border-radius: 0;
  background-color: transparent;
  min-height: 38px;
}

.c0 .multiselect__control--is-focused {
  box-shadow: none;
}

.c0 .multiselect__value-container {
  padding: 5px 5px 5px 0;
}

.c0 .multiselect__menu {
  border-radius: 2px;
}

<StateManager
  className="c0"
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
/>
`;
